### PR TITLE
feat(search): add search endpoints

### PR DESCRIPTION
**Pull request by jtheoof**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: go
+go:
+- "1.X"
+script:
```
Comment (daddykotex) [inline .travis.yml]: can you make sure that travis runs `gofmt` and fails if it leaves files modified (a.k.a the commits in the PR contains unformatted code)

**Comment by jtheoof** [comment]
@daddykotex Thanks. I can do it for the `NewClient` with `id` and `secret` but for `NewClientWithToken` I cannot do it since I don't have any other way to authenticate the client. Is that acceptable?

File: .travis.yml (hunk 1 above)
Comment (jtheoof) [reply .travis.yml]: I'll check how I can do that

**Comment by daddykotex** [comment]
That's exactly what I meant 👍

File: .travis.yml (hunk 1 above)
Comment (daddykotex) [reply .travis.yml]: This looks good

File: mnubo/search_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	var results = [2]SearchResults{}
+	cases := []struct {
+		Error error
+	}{
+		{
+			Error: m.createBasicQuery(SimpleQuery{
+				From: "event",
+				Select: []SelectOperation{
+					{
+						Count: "*",
```
Comment (dmasse76) [inline mnubo/search_test.go]: How would you recommend handling dates?  As as string?
```
"where": {
    "and": [
      {
        "x_timestamp": {
          "dateBetween": "2018-10-02T04:00:00.000Z",
          "and": "2018-10-11T19:09:59.000Z"
        }
      }
    ]
  }
```

**Pull request merged**